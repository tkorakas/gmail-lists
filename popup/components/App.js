import React, {Component} from 'react';
import List from './List';
import RecipientsList from './RecepientsList';

export default class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      page: 'list',
      item: ''
    };

    this.openDonateButton = this.openDonateButton.bind(this);
    this.changePage = this.changePage.bind(this);
    this.renderPage = this.renderPage.bind(this);
  }

  /**
   * Redirect to donate button.
   */
  openDonateButton(e) {
    chrome.tabs.create({url: 'https://www.paypal.me/tkorakas/2'});
  }

  /**
   * Change page.
   *
   * @param page
   *  Page to change to.
   * @param item
   *  If page equals to recipients needs an item.
   */
  changePage(item = '') {
    const page = this.state.page === 'list' ? 'recipients' : 'list';

    this.setState({
      page,
      item
    });
  }

  /**
   * Render List or recipients page based on state,
   */
  renderPage() {
    return this.state.page === 'list' ? <List changePage={this.changePage}/> :
      <RecipientsList item={this.state.item} changePage={this.changePage}/>
  }

  render() {
    return (
      <div>
        {this.renderPage()}
        <footer className="donate-section">
          <a onClick={this.openDonateButton} href="">Donate a beer</a>
        </footer>
      </div>
    );
  }
}
