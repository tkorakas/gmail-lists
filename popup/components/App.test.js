import React from 'react';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() });
import renderer from 'react-test-renderer';
import chrome from '../../utils/ChromeMock';
import App from './App';


describe('App component', () => {
  beforeEach(() => {
    window.chrome = chrome;
  });

  test('snapshot test', () => {
    const tree = renderer
    .create(<App />)
    .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('change page', () => {
    // const app = mount(
    //   <App />
    // );
  });
});
