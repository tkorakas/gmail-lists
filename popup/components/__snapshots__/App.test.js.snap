// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component snapshot test 1`] = `
<div>
  <div
    className="app-list"
  >
    <span
      className="input-container"
    >
      <div
        className="MuiInput-root-1 MuiInput-fullWidth-8 MuiInput-underline-5"
        onKeyPress={[Function]}
        style={
          Object {
            "paddingLeft": 4,
          }
        }
      >
        <input
          autoFocus={true}
          className="MuiInput-input-9"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Create new list"
          type="text"
        />
      </div>
    </span>
    <nav
      className="MuiList-root-14 MuiList-padding-15"
    />
  </div>
</div>
`;
